src/wovnio/html/HtmlConverter.php

```diff
@@ -74,6 +82,49 @@ private function insertSnippet($dom) {
 
     $token = $this->token;
     $insert_tags = array('head', 'body', 'html');
+    foreach($insert_tags as $tag_name) {
+      $parents = $dom->find($tag_name);
+      if (count($parents) > 0) {
+        $parent = $parents[0];
+        $parent->innertext = "<script src=\"//j.wovn.io/1\" data-wovnio=\"key=$token\" data-wovnio-type=\"backend_without_api\" async></script>" . $parent->innertext;
+        return;
+      }
+    }
+  }
+
+  /**
+   * Insert hreflang tags for all supported_langs
+   *
+   * @param simple_html_dom_node $dom
+   */
+  public function insertHreflangTags($dom) {
+    $lang_codes = $this->store->settings['supported_langs'];
+    foreach ($dom->find('link') as $node) {
+      $hreflangValue = $node->getAttribute('hreflang');
+      if (in_array(Lang::getCode($hreflangValue), $lang_codes)) {
+        // remove node
+        $node->outertext = '';
+      }
+    }
+
+    $insert_tags = array('head', 'body', 'html');
+    foreach($insert_tags as $tag_name) {
+      $parents = $dom->find($tag_name);
+      if (count($parents) > 0) {
+        $parent = $parents[0];
+        $hreflangTags = array('');
+        foreach ($lang_codes as $lang_code) {
+          $href = Url::addLangCode($this->headers->url, $this->store, $lang_code, $this->headers);
+          array_push($hreflangTags, '<link rel="alternate" hreflang="' . Lang::iso639_1Normalization($lang_code) . '" href="' . $href . '">');
+        }
+        $parent->innertext = implode("\n", $hreflangTags) . "\n" . $parent->innertext;
```

Comment:
I think that it is better using `appendChild` instead of `innertext`.

Reply:
There is no appendChild. Also, we are copying the above function that Hiroaki wrote.
http://simplehtmldom.sourceforge.net/manual_api.htm

Reply:
Okay.